### PR TITLE
i18n(fr) Update and translated `imports.mdx`

### PR DESCRIPTION
**Pull request by McFlyPartages**
#### Description (required)
Update and translated imports.mdx

### CODE REVIEW

**Comment by McFlyPartages** [comment]
J'ai remis les liens de la ligne 93 et 153 en anglais, car la doc n'est pas encore traduite mais en cours #4948

File: src/content/docs/fr/guides/imports.mdx [hunk 1]
```diff
@@ -1,38 +1,46 @@
...
-- Packages NPM
+- Paquets NPM
 - JSON (`.json`)
 - JSX (`.jsx`, `.tsx`)
 - CSS (`.css`)
 - Modules CSS (`.module.css`)
-- Images & éléments (`.svg`, `.jpg`, `.png`, etc.)
+- Images et actifs (`.svg`, `.jpg`, `.png`, etc.)
+
+De plus, vous pouvez étendre Astro pour ajouter le support de différents [Frameworks UI ](/fr/core-concepts/framework-components/) comme React, Svelte et Vue. Vous pouvez également installer l'[intégration Astro MDX](/fr/guides/integrations-guide/mdx/) et utiliser les fichiers `.mdx` dans votre projet.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:27]: 429 Too many _space_
```suggestion
De plus, vous pouvez étendre Astro pour ajouter le support de différents [Frameworks UI](/fr/core-concepts/framework-components/) comme React, Svelte et Vue. Vous pouvez également installer l'[intégration Astro MDX](/fr/guides/integrations-guide/mdx/) et utiliser les fichiers `.mdx` dans votre projet.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 2]
```diff
@@ -1,38 +1,46 @@
...
+
+### Les fichiers dans `public/`
 
-Si vous ne trouvez pas le type de ressource que vous recherchez, consultez notre [Bibliothèque d'intégrations](https://astro.build/integrations/). Vous pouvez étendre Astro pour ajouter la prise en charge de différents types de fichiers, comme les composants Svelte et Vue.
+Vous pouvez placer n'importe quelle ressource statique dans le répertoire [`public/`](/fr/core-concepts/project-structure/#public) de votre projet, et Astro le copiera directement dans votre version finale sans y toucher. Les fichiers dans le répertoire `public/` ne sont pas construits ou regroupés par Astro, ce qui signifie que n'importe quel type de fichier est supporté. Vous pouvez référencer un fichier `public/` par un chemin URL directement dans vos modèles HTML.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:31]: `par un chemin URL` sounds weird. I don't really have other suggestion, so don't take mine but I don't know if you have any idea.

```suggestion
Vous pouvez placer n'importe quelle ressource statique dans le répertoire [`public/`](/fr/core-concepts/project-structure/#public) de votre projet, et Astro le copiera directement dans votre version finale sans y toucher. Les fichiers dans le répertoire `public/` ne sont pas construits ou regroupés par Astro, ce qui signifie que n'importe quel type de fichier est supporté. Vous pouvez référencer un fichier `public/` par un chemin directement dans vos modèles HTML.
```

or maybe:
```suggestion
Vous pouvez placer n'importe quelle ressource statique dans le répertoire [`public/`](/fr/core-concepts/project-structure/#public) de votre projet, et Astro le copiera directement dans votre version finale sans y toucher. Les fichiers dans le répertoire `public/` ne sont pas construits ou regroupés par Astro, ce qui signifie que n'importe quel type de fichier est supporté. Vous pouvez référencer un fichier `public/` par un chemin d'URL directement dans vos modèles HTML.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 3]
```diff
@@ -41,11 +49,20 @@ import { getUser } from './user.ts';
...
+Astro comprend un support intégré pour [TypeScript] (https://www.typescriptlang.org/). Vous pouvez importer des fichiers `.ts` et `.tsx` directement dans votre projet Astro, et même écrire du code TypeScript directement dans votre [script de composant Astro](/fr/core-concepts/astro-components/#le-script-du-composant) et n'importe quelle [balise de scripts](/fr/guides/client-side-scripts/).
+
+**Astro n'effectue aucune vérification de type lui-même.** La vérification de type doit être prise en charge en dehors d'Astro, soit par votre IDE, soit par un script séparé. Pour vérifier le type des fichiers Astro, la commande [`astro check`](/fr/reference/cli-reference/#astro-check) est fournie.
+
+:::note[TypeScript et les extensions de fichiers]
+Selon les [règles de résolution des modules TypeScript](https://www.typescriptlang.org/docs/handbook/module-resolution.html), les extensions de fichiers `.ts` et `.tsx` ne doivent pas être utilisées lors de l'importation de fichiers TypeScript. A la place, utilisez les extensions de fichiers `.js`/`.jsx` ou omettez complètement l'extension de fichier.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:57]: I'm sorry for that :sweat_smile: :
```suggestion
Selon les [règles de résolution des modules TypeScript](https://www.typescriptlang.org/docs/handbook/module-resolution.html), les extensions de fichiers `.ts` et `.tsx` ne doivent pas être utilisées lors de l'importation de fichiers TypeScript. À la place, utilisez les extensions de fichiers `.js`/`.jsx` ou omettez complètement l'extension de fichier.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 4]
```diff
@@ -55,21 +72,26 @@ import { MyComponent } from './MyComponent.jsx';
...
 :::
 
-## Packages NPM
+## Paquets NPM
 
-```js
-// Retourne les packages NPM React & React-DOM
-import React from 'react';
-import ReactDOM from 'react-dom';
+Si vous avez installé un paquetage NPM, vous pouvez l'importer dans Astro.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:83]: I'm not shure for the word "paquetage"
```suggestion
Si vous avez installé un paquet NPM, vous pouvez l'importer dans Astro.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 5]
```diff
@@ -87,9 +109,7 @@ Astro prend en charge l'importation de fichiers JSON directement dans votre appl
 import './style.css';
 ```
 
-Astro supporte l'importation de fichiers CSS directement dans votre application. Les styles importés n'exposent aucune exportation, mais l'importation d'un style ajoutera automatiquement ces styles à la page. Cela fonctionne pour tous les fichiers CSS par défaut, et peut prendre en charge les langages de compilation vers le CSS comme Sass et Less via des plugins.
-
-Si vous préférez ne pas écrire de CSS, Astro prend également en charge toutes les bibliothèques CSS-in-JS populaires (ex : styled-components) pour le style.
+Astro prend en charge l'importation de fichiers CSS directement dans votre application. Les styles importés n'exposent à aucune exportation, mais l'importation d'un style ajoutera automatiquement ce style à la page. Cela fonctionne pour tous les fichiers CSS par défaut, et peut prendre en charge les langages CSS compilés comme Sass et Less via des plugins..
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:112]: I also have a problem with this sentence: `Les styles importés n'exposent à aucune exportation`. Maybe with `ne nécessitent pas d'exportation` that could be better.
But anyway, you duplicate the final dot `..`
```suggestion
Astro prend en charge l'importation de fichiers CSS directement dans votre application. Les styles importés ne nécessitent pas d'exportations, mais l'importation d'un style ajoutera automatiquement ce style à la page. Cela fonctionne pour tous les fichiers CSS par défaut, et peut prendre en charge les langages CSS compilés comme Sass et Less via des plugins.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 6]
```diff
@@ -102,47 +122,103 @@ import styles from './style.module.css';
...
+import svgReference from './image.svg'; // svgReference === '/src/image.svg'
+import txtReference from './words.txt'; // txtReference === '/src/words.txt'
 
 // Cet exemple utilise JSX, mais vous pouvez utiliser des références d'importation avec n'importe quel Framework.
-<img src={imgReference} />;
+<img src={imgReference.src} alt="image description" />;
 ```
 
-Toutes les autres ressources qui ne sont pas explicitement mentionnés ci-dessus peuvent être importés via ESM `import` et retourneront une référence URL vers la ressource finale construit. Cela peut être utile pour référencer des ressources non-JS par URL, comme créer un élément image avec un attribut `src` pointant vers cette image.
+Toutes les autres ressources qui ne sont pas explicitement mentionnés ci-dessus peuvent être importés via ESM `import` et renverront une référence URL à la ressource finale construite. Cela peut être utile pour référencer des ressources non-JS par URL, comme créer un élément image avec un attribut `src` pointant vers cette image.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:140]:
```suggestion
Toutes les autres ressources qui ne sont pas explicitement mentionnés ci-dessus peuvent être importés via l'`import` ESM et renverront une référence URL à la ressource finale construite. Cela peut être utile pour référencer des ressources non-JS par URL, comme créer un élément image avec un attribut `src` pointant vers cette image.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 7]
```diff
@@ -102,47 +122,103 @@ import styles from './style.module.css';
...
+  </article>
+))}
+</div>
+```
+
+Les composants Astro importés avec `Astro.glob` sont de type [`AstroInstance`](/fr/reference/api-reference/#astro-files). Vous pouvez effectuer le rendu de chaque instance de composant en utilisant sa propriété `default` :
+
+```astro title="src/pages/component-library.astro" {8}
+---
+// imports all files that end with `.astro` in `./src/components/`
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:175]:
```suggestion
// importe tous les fichiers qui se terminent par `.astro` dans `./src/components/`
```

File: src/content/docs/fr/guides/imports.mdx [hunk 8]
```diff
@@ -102,47 +122,103 @@ import styles from './style.module.css';
...
+const components = await Astro.glob('../components/*.astro');
+---
+<!-- Display all of our components -->
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:178]:
```suggestion
<!-- Affiche tous nos composants -->
```

File: src/content/docs/fr/guides/imports.mdx [hunk 9]
```diff
@@ -102,47 +122,103 @@ import styles from './style.module.css';
...
+
 ### Patterns globaux
 
 Un pattern global est un chemin d'accès à un fichier qui prend en charge les caractères génériques spéciaux. Il est utilisé pour référencer plusieurs fichiers dans votre projet en même temps.
 
 Par exemple, le pattern global `./pages/**/*.{md,mdx}` commence dans le sous-répertoire pages, et regarde dans tous ses sous-répertoires (`/**`), et correspond à tous les noms de fichier (`/*`) qui finissent soit par `.md` ou `.mdx` (`.{md,mdx}`).
 
+#### Patterns globaux dans Astro
+
+Pour être utilisé avec `Astro.glob()`, le pattern globale doit être une chaîne littérale et ne peut pas contenir de variables. Voir [le guide de dépannage](/fr/guides/troubleshooting/#astroglob---no-matches-found) pour une solution de contournement.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:194]:
```suggestion
Pour être utilisé avec `Astro.glob()`, le pattern global doit être une chaîne littérale et ne peut pas contenir de variables. Voir [le guide de dépannage](/fr/guides/troubleshooting/#astroglob---no-matches-found) pour une solution de contournement.
```

File: src/content/docs/fr/guides/imports.mdx [hunk 10]
```diff
@@ -156,3 +232,13 @@ const data = JSON.parse(json);
 
 <span>Version : {data.version}</span>
 ```
+
+## Extension de la prise en charge des types de fichiers
+
+Avec **Vite** et les plugins **Rollup** compatibles, vous pouvez importer des types de fichiers qui ne sont pas pris en charge nativement par Astro. Pour savoir où trouver les plugins dont vous avez besoin, consultez la section [Finding Plugins](https://vitejs.dev/guide/using-plugins.html#finding-plugins) de la documentation Vite.
+
+:::note[Configuration du plugin]
+Referez-vous à la documentation de votre plugin pour connaître les options de configuration et savoir comment l'installer correctement.
```
Comment (pioupia) [inline src/content/docs/fr/guides/imports.mdx:241]: With some French accent, that's always better!
```suggestion
Référez-vous à la documentation de votre plugin pour connaître les options de configuration et savoir comment l'installer correctement.
```

**Review by pioupia — CHANGES_REQUESTED** [review]

**Comment by McFlyPartages** [comment]
PR mise a jour avec vos suggestions

**Review by pioupia — APPROVED** [review]

File: src/content/docs/fr/guides/imports.mdx [hunk 11]
```diff
@@ -102,47 +122,103 @@ import styles from './style.module.css';
...
+
 ## WASM
 
 ```js
 // Charge et initialise le fichier WASM demandé
 const wasm = await WebAssembly.instantiateStreaming(fetch('/example.wasm'));
 ```
 
-Astro supporte le chargement de fichiers WASM directement dans votre application en utilisant l'API [`WebAssembly`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) du navigateur.
+Astro permet de charger des fichiers WASM directement dans votre application en utilisant l'API [`WebAssembly`] (https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) du navigateur.
```
Comment (Yan-Thomas) [inline src/content/docs/fr/guides/imports.mdx:215]:
```suggestion
Astro permet de charger des fichiers WASM directement dans votre application en utilisant l'API [`WebAssembly`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly) du navigateur.
```

**Review by Yan-Thomas — APPROVED** [review]
LGTM! There was a broken link but I fixed it myself 🙌

**Pull request merged**